content/realtime/delta.textile

```diff
@@ -0,0 +1,222 @@
+
+On some occasions a channel subscriber can experience a discontinuity in the sequence of messages it receives on any given channel. There are several possible reasons for this: the connection can drop, and there will be a discontinuity of the client is unable to reconnect within the two-minute window it is allowed to preserve connection continuity; the outbound connection might have been rate-limited, which causes some messages to be dropped; or there might have been some internal error in the Ably system which leads to the server being unable to preserve continuity on the channel. In these cases, the service indicates the discontinuity to the client, together with the reason, and this is usually visible to the subscriber in a channel @UPDATE@ event. If a subscriber has a delta-mode subscription and the channel in question experiences a discontunity, then a non-delta message will be delivered to the client as the first message after the discontinuity, so that lost messages do not prevent the client from reconstituting messages from deltas.
+
+h2(#using-deltas). Using deltas
+
+h3(#using-deltas-12). Via an Ably library from v1.2 (via the delta channel parameter)
+
+The most common way to subscribe to Ably channels is via a realtime connection, using an Ably realtime library.
+
+From version 1.2, Ably libraries support the ability to subscribe to a channel in delta mode. For many libraries this requires no change on the part of the caller except to specify the [delta channel param](./channel-params) when subscribing to the channel. In some libraries, the @vcdiff@ delta decoding library is excluded from the default library distribution in order to avoid bloating the library; in these cases, it is also necessary to supply the delta decoder plugin when instancing the Ably library.
+
+
+```[jsall]
+  const vcdiffPlugin = require('{{vcdiff-plugin}}')
+  const realtime = new Ably.Realtime({key: '{{API_KEY}}', plugins: {vcdiffDecoder: vcdiffDecoder}});
+  realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', {
+    delta: 'vcdiff'
+  }).subscribe(msg => console.log("Received message: ", msg));
+```
+
+```[java]
+  AblyRealtime ably = new AblyRealtime("{{API_KEY}}"")
+  Channel channel = ably.channels.get("{{RANDOM_CHANNEL_NAME}}", new ChannelOptions{{params = Map.of("delta", "vcdiff")}});
+  channel.subscribe(new MessageListener() {
+    @Override
+    public void onMessage(Message message) {
+      System.out.println("Received `" + message.name + "` message with data: " + message.data);
+    }
+  });
+```
+
+```[obj-c]
+TBD
+```
+
+```[dotnet]
+TBD
+```
+
+h3(#using-deltas-11). Via an Ably library before v1.2 (via a qualified channel name)
```

Comment:
How are they going to apply the deltas in the pre 1.2 libs? In theory they could since the decoding will fail at vcdiff and the actual delta will get returned. Then they can use the OS libs we have to decode that. But that has to be documented maybe or there has to be reference to the non-ably usage section below.